hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/CertificateClient.java

```diff
@@ -174,15 +175,9 @@ default void assertValidKeysAndCertificate() throws OzoneSecurityException {
     }
   }
 
-  /**
-   * Return the store factory for key manager and trust manager for server.
-   */
-  KeyStoresFactory getServerKeyStoresFactory() throws CertificateException;
+  KeyManager getKeyManager() throws CertificateException;
```

Comment:
If you define the return value of this method to be ReloadingX509KeyManager, then you can spare a cast in the methods implementing these and in some tests also, while that way the returned object still remains a KeyManager also, so it can be set to TLSConfig, and SSLContext and other places as is. This same is true for the getTrustManager method where you can use ReloadingX509TrustManager instead of TrustManager.

Reply:
Thank for that, that was a great observation. Fixed it